### ISSUE TITLE
CarrierWave::Storage::constants => [:Abstract, :File], no fog

### ISSUE THREAD

**Issue by Nephos** [carrierwaveuploader/carrierwave#1870]
Hi,

I have the following issue, only on the master branch

**Using carrierwave 0.10.0 from git://github.com/carrierwaveuploader/carrierwave.git (at master@aa74e9e)**
```text
CarrierWave::Storage::constants
=> [:Abstract, :File]
```
*Note: same issue with older revision like 1e4e7b0, despite an other of my project works with*

**Using carrierwave 0.10.0**
```text
CarrierWave::Storage::constants
=> [:Abstract, :File, :Fog]
```

**Gemfile**
```text
gem 'carrierwave', github: 'carrierwaveuploader/carrierwave' # no Fog :(
#gem 'carrierwave' # no mount_uploaderS :(
gem 'fog'
gem 'fog-aws'
```

**Gemfile.lock**
```text
GIT
  remote: git://github.com/carrierwaveuploader/carrierwave.git
  revision: 1e4e7b0bf20fee70d8ac3815d0921144f39e14cf
  ref: 1e4e7b0bf20fee70d8ac3815d0921144f39e14cf
  specs:
    carrierwave (0.10.0)
      activemodel (>= 3.2.0)
      activesupport (>= 3.2.0)
      json (>= 1.7)
      mime-types (>= 1.16)

# ...

PLATFORMS
  ruby

DEPENDENCIES
  byebug
  carrierwave!
  coffee-rails (~> 4.1.0)
  fog
  fog-aws
  jbuilder (~> 2.0)
  jquery-rails
  rails (= 4.2.5)
  sass-rails (~> 5.0)
  sdoc (~> 0.4.0)
  spring
  sqlite3
  turbolinks
  uglifier (>= 1.3.0)
  web-console (~> 2.0)

BUNDLED WITH
   1.11.2
````

**Comment by mehlah** [comment]
Beware the code in the master branch is not yet released and quite different from 0.10.
Especially, the way Fog is loaded has changed.
Now Fog is eager loaded as you can see at https://github.com/carrierwaveuploader/carrierwave/blob/3cabb21b9a70309ad5617f33a338c07fc64c834a/lib/carrierwave/uploader/configuration.rb#L110-L115

This means CarrierWave will load it if `fog_credentials` hash exists. You should have that in your CarrierWave initializer.
Here is a gist with a test proving that, for the current master branch of CarrierWave https://gist.github.com/mehlah/b504e8d989fb0c37d6d2

Hope this helps

@thomasfedb I guess this could be closed :)

**Comment by thomasfedb** [comment]
Thanks @mehlah
